eclipse-collections/src/main/java/org/eclipse/collections/impl/collector/Collectors2.java

```diff
@@ -50,12 +62,40 @@ private Collectors2()
                 EMPTY_CHARACTERISTICS);
     }
 
-    public static <T> Collector<T, ?, MutableStack<T>> toStack()
+    public static <T> Collector<T, ?, MutableList<T>> toList()
     {
         return Collector.of(
-                Stacks.mutable::empty,
-                MutableStack::push,
-                (r1, r2) -> {r2.each(r1::push); return r1;},
+                Lists.mutable::empty,
+                MutableList::add,
+                MutableList::withAll,
+                Collector.Characteristics.IDENTITY_FINISH);
+    }
+
+    public static <T> Collector<T, ?, MutableSet<T>> toSet()
+    {
+        return Collector.of(
+                Sets.mutable::empty,
+                MutableSet::add,
+                MutableSet::withAll,
+                Collector.Characteristics.UNORDERED);
+    }
+
+    public static <T> Collector<T, ?, MutableBag<T>> toBag()
+    {
+        return Collector.of(
+                Bags.mutable::empty,
+                MutableBag::add,
+                MutableBag::withAll,
+                Collector.Characteristics.UNORDERED);
+    }
+
+    public static <T> Collector<T, MutableList<T>, MutableStack<T>> toStack()
+    {
+        return Collector.of(
+                Lists.mutable::empty,
+                MutableList::add,
+                MutableList::withAll,
+                Stacks.mutable::ofAll,
                 EMPTY_CHARACTERISTICS);
```

Comment:
I see you edited toStack() further and my previous comment doesn't apply anymore, but I'm still curious about test coverage.